File: Ejemplo03/app/src/main/java/com/naldana/ejemplo03/MainActivity.java
package com.naldana.ejemplo03;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private EditText mNameEditText;
    private TextView mDisplayNameTextView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mNameEditText = findViewById(R.id.et_name);
        mDisplayNameTextView = findViewById(R.id.tv_display_name);
    }


    public void displayName(View v){
        String name = mNameEditText.getText().toString();

        if(name.isEmpty()) {
            mDisplayNameTextView.setText(R.string.hint_text_instrucction_name);
        } else {
            mDisplayNameTextView.setText(name);
        }

    }
}

File: Ejemplo08/app/src/main/java/com/naldana/ejemplo08/MainActivity.kt
package com.naldana.ejemplo08

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.GridLayoutManager
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import com.naldana.ejemplo08.models.Pokemon
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private lateinit var viewAdapter: PokemonAdapter
    private lateinit var viewManager: RecyclerView.LayoutManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initRecycler()
    }


    fun initRecycler() {

        var pokemon: MutableList<Pokemon> = MutableList(100) {i ->
            Pokemon(i,"Name" + i, "Tyep " + i)
        }

        viewManager = LinearLayoutManager(this)

        viewAdapter = PokemonAdapter(pokemon)

        rv_pokemon_list.apply {
            setHasFixedSize(true)
            layoutManager = viewManager
            adapter = viewAdapter
        }

    }
}
